### ISSUE TITLE
Check that explicit protocol subclass defines abstract methods

### ISSUE THREAD

**Issue by tekumara** [microsoft/pyright#1498]
Would be great if pyright checks that an explicit subclass of a protocol implements abstract methods. 
 
eg: pyright will produce no errors on:

```python
from typing import Protocol, Tuple
from abc import ABCMeta, abstractmethod

class RGB(Protocol):
    rgb: Tuple[int, int, int]

    @abstractmethod
    def intensity(self) -> int:
        return 0

class Point(RGB):
    def __init__(self, red: int, green: int, blue: int) -> None:
        self.rgb = red, green, blue

    # Type checker might warn that 'intensity' is not defined

p = Point(1,2,3)
```

mypy will error with:
```
test.py:17: error: Cannot instantiate abstract class 'Point' with abstract attribute 'intensity'
```

If `RGB` is an ABC, ie: `class RGB(metaclass=ABCMeta)` then pyright will error with:
```
  17:5 - error: Cannot instantiate abstract class "Point"
    "RGB.intensity" is abstract (reportGeneralTypeIssues)
```
So ABCs are more useful here. Would be nice to make explicit Protocol subtypes as useful.

pyright 1.1.107
python 3.8.6

**Comment by erictraut** [comment]
Good suggestion. This will be implemented in the next release.

**Comment by erictraut** [comment]
This is implemented in Pyright 1.1.112, which I just published. It will also be included in the next release of Pylance.
